contracts/pools/aeth/StableSwapAETH.vy

```diff
@@ -0,0 +1,857 @@
+    dy_fee: uint256 = dy * self.fee / FEE_DENOMINATOR
+    dy_admin_fee: uint256 = dy_fee * self.admin_fee / FEE_DENOMINATOR
+
+    self.balances[i] = x * PRECISION / rates[i]
+    self.balances[j] = (y + (dy_fee - dy_admin_fee)) * PRECISION / rates[j]
+
+    dy = (dy - dy_fee) * PRECISION / rates[j]
+    assert dy >= min_dy, "Exchange resulted in fewer coins than expected"
+
+    coin: address = self.coins[1]
+    if i == 0:
+        assert msg.value == dx
+        assert ERC20(coin).transfer(msg.sender, dy)
+    else:
+        assert msg.value == 0
+        assert ERC20(coin).transferFrom(msg.sender, self, dx)
+        raw_call(msg.sender, b"", value=dy)
+
+    log TokenExchange(msg.sender, i, dx, j, dy)
+
+    return dy
+
+
+@external
+@nonreentrant('lock')
+def remove_liquidity(_amount: uint256, min_amounts: uint256[N_COINS]) -> uint256[N_COINS]:
+    """
+    @notice Withdraw coins from the pool
+    @dev Withdrawal amounts are based on current deposit ratios
+    @param _amount Quantity of LP tokens to burn in the withdrawal
+    @param min_amounts Minimum amounts of underlying coins to receive
+    @return List of amounts of coins that were withdrawn
+    """
+    _lp_token: address = self.lp_token
+    total_supply: uint256 = ERC20(_lp_token).totalSupply()
+    amounts: uint256[N_COINS] = empty(uint256[N_COINS])
+    fees: uint256[N_COINS] = empty(uint256[N_COINS])
+
+    for i in range(N_COINS):
+        _balance: uint256 = self.balances[i]
+        value: uint256 = _balance * _amount / total_supply
+        assert value >= min_amounts[i], "Withdrawal resulted in fewer coins than expected"
+        self.balances[i] = _balance - value
+        amounts[i] = value
+        if i == 0:
+            raw_call(msg.sender, b"", value=value)
+        else:
+            assert ERC20(self.coins[1]).transfer(msg.sender, value)
+
+    CurveToken(_lp_token).burnFrom(msg.sender, _amount)  # Will raise if not enough
+
+    log RemoveLiquidity(msg.sender, amounts, fees, total_supply - _amount)
+
+    return amounts
+
+
+@external
+@nonreentrant('lock')
+def remove_liquidity_imbalance(amounts: uint256[N_COINS], max_burn_amount: uint256) -> uint256:
+    """
+    @notice Withdraw coins from the pool in an imbalanced amount
+    @param amounts List of amounts of underlying coins to withdraw
+    @param max_burn_amount Maximum amount of LP token to burn in the withdrawal
+    @return Actual amount of the LP token burned in the withdrawal
+    """
+    assert not self.is_killed
+    amp: uint256 = self._A()
+    rates: uint256[N_COINS] = self._stored_rates()
+    old_balances: uint256[N_COINS] = self.balances
+    D0: uint256 = self.get_D_mem(rates, old_balances, amp)
+
+    new_balances: uint256[N_COINS] = old_balances
+    for i in range(N_COINS):
+        new_balances[i] -= amounts[i]
+    D1: uint256 = self.get_D_mem(rates, new_balances, amp)
+
+    _lp_token: address = self.lp_token
+    token_supply: uint256 = ERC20(_lp_token).totalSupply()
```

Comment:
the check against `token_supply > 0` can be removed, because the math later in this function will revert from divide-by-zero.  as such, `token_supply` can then be declared later in the code (after the for loop)